package-res/ccc/core/cartesian/axis/abstract-cart-axis-panel.js

```diff
@@ -1553,3 +1510,281 @@ def
+ *                                       rotatethenhide Rotate the labels so they don't overlap. If not possible,
+ *                                                      hides the labels that still overlap.
+ *
+ * @param {int} labelRotationDirection  Controls the rotation direction of the labels. Possible values:
+ *                                        1             Clockwise rotation (default).
+ *                                       -1             Counterclockwise rotation. Inverts the signal of the
+ *                                                      rotation angle.
+ *
+ * @param {array} labelDesiredAngles    List of angles to try when overlappedLabelsMode is 'rotate' or 'rotatethenhide'.
+ *                                      The nearest above the minimum non-overlapping angle is chosen, or just the
+ *                                      nearest if none is bigger.
+ *
+ * @param {Number} distanceBetweenTicks The amount of separation between the label anchors.
+ *
+ * @param {Number} labelSpacingMin      Minimum separation space that the user wants between the closest edges of the
+ *                                      bounding boxes of two consecutive labels, measured perpendicularly to the label
+ *                                      text direction. Value in em.
+ *
+ * @param {Number} fontPxWidth          Additional minimum horizontal space between labels, derived from the font's
+ *                                      'x' character width.
+ *
+ * @param {String} axisAnchor           The axis position. Possible values:
+ *                                       top, bottom, left, right.
+ *
+ * @param {Object} layoutInfo           In/out parameter with the current layout info.
+ *
+ *                                      The following properties are read:
+ *                                       textHeight:         The height of the labels.
+ *                                       maxTextWidth:       The maximum label width.
+ *                                       textAngle:          The label rotation angle (when overlappedLabelsMode isn't
+ *                                                           'rotate' or 'rotatethenhide').
+ *                                       ticks.length:       The axis total number of ticks.
+ *
+ *                                      This function always adds these properties:
+ *                                       tickVisibilityStep: The visibility step of the labels.
+ *                                       lockedTextAngle:    The text angle to apply to the labels.
+ *
+ *                                      Additionally, if overlappedLabelsMode is 'rotate' or 'rotatethenhide', these
+ *                                      properties are also defined:
+ *                                       lockedTextAlign:    The horizontal label anchor.
+ *                                       lockedTextBaseline: The vertical label anchor.
```

Comment:
Following the naming used in similar cases, like `beginLocked`, `leftLocked`, `minLocked`, etc.. , these would be named `textAlignLocked` and `textBaselineLocked`.